### PR TITLE
Cleanup obsolete parts of simple istio-agent xds server

### PR DESCRIPTION
**Pull request by howardjohn**
This is only used for SDS agent, so simplify it to only do what it needs

### CODE REVIEW

**Review by zirain — APPROVED** [review]

**Review by ramaraochavali — APPROVED** [review]

File: security/pkg/nodeagent/sds/sdsservice.go [hunk 1]
```diff
@@ -58,14 +62,25 @@ type sdsservice struct {
...
-	s := xds.NewXDS(stop)
+	// Setup a mostly empty NewEnvironment
+	env := model.NewEnvironment()
+	env.ServiceDiscovery = aggregate.NewController(aggregate.Options{})
+	env.ConfigStore = memory.Make(collection.NewSchemasBuilder().Build())
+	env.Watcher = meshcfg.NewFixedWatcher(meshcfg.DefaultMeshConfig())
+	env.PushContext().Mesh = env.Watcher.Mesh()
+	env.Init()
+
+	s := xds.NewDiscoveryServer(env, "istiod", "", map[string]string{})
```
Comment (hzxuzhonghu) [inline security/pkg/nodeagent/sds/sdsservice.go:76]: nit: This is the node agent, so why istiod not leave it to empty

File: security/pkg/nodeagent/sds/sdsservice.go [hunk 2]
```diff
@@ -58,14 +62,25 @@ type sdsservice struct {
...
+// The DiscoveryServer was originally designed for Istiod, so a lot of this is plugging in fake empty information.
 func NewXdsServer(stop chan struct{}, gen model.XdsResourceGenerator) *xds.DiscoveryServer {
-	s := xds.NewXDS(stop)
+	// Setup a mostly empty NewEnvironment
+	env := model.NewEnvironment()
+	env.ServiceDiscovery = aggregate.NewController(aggregate.Options{})
+	env.ConfigStore = memory.Make(collection.NewSchemasBuilder().Build())
+	env.Watcher = meshcfg.NewFixedWatcher(meshcfg.DefaultMeshConfig())
+	env.PushContext().Mesh = env.Watcher.Mesh()
+	env.Init()
```
Comment (hzxuzhonghu) [inline security/pkg/nodeagent/sds/sdsservice.go:74]: move out, maybe we need it for unit test in future

Comment (howardjohn) [reply security/pkg/nodeagent/sds/sdsservice.go:74]: Unit tests use FakeXdsServer already

File: security/pkg/nodeagent/sds/sdsservice.go (hunk 1 above)
Comment (howardjohn) [reply security/pkg/nodeagent/sds/sdsservice.go]: this is unused entirely, I will clean up. And cluster ID as well

**Pull request merged**